### PR TITLE
Add repository for masques

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes some part of #3309

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [ ] ~Do all end-to-end tests pass?~
* [ ] ~Any other testing performed?~
    > ~Tested by {Running the Monkey locally with relevant config/running Island/...}~
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py [hunk 1]
```diff
@@ -14,12 +13,18 @@
 )
 from common.types import AgentID, Event
 
+logger = logging.getLogger(__name__)
+
+try:
+    import pwd
+except ModuleNotFoundError as err:
+    logger.debug(f"Most likely running on an unsupported OS: {err}")
```
Comment (ilija-lazoroski) [inline monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py:21]: Do we need to do this in the plugin? The plugin is not supported on Windows so I don't see how it is an issue.

Comment (shreyamalviya) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py:21]: Read the commit message. 
https://github.com/guardicore/monkey/pull/3331/commits/831ad89526bbe2c4094718231384167eb1805aa5

Comment (shreyamalviya) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py:21]: We do.
Read the commit message. https://github.com/guardicore/monkey/pull/3331/commits/831ad89526bbe2c4094718231384167eb1805aa5

Comment (shreyamalviya) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py]: Found a better way. See [41ea0e6](https://github.com/guardicore/monkey/pull/3331/commits/41ea0e62ad80b88d80336a1f92d3ed8937ff2514).

Comment (ilija-lazoroski) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py]: Nice job!

Comment (mssalvatore) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py]: Can we just @pytest.skip the test and the fixture? Does @pytest.skip work on fixtures?

Comment (shreyamalviya) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py]: We'd be skipping both tests anyway. What's wrong with skipping the whole file?

Comment (mssalvatore) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py]: Skipping the whole file makes sense. My main objection is all of the ignores for the type checker. But it's not really a big deal.

Would this work?
```
pytest.importskip("pwd")
import pwd
```

File: monkey/monkey_island/cc/repositories/i_masquerade_repository.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        Retrieve the masque for the specified OS.
+
+        :param operating_system: The OS whose masque is to be retrieved
+        :return: The masque for the specified OS
+        :raises RetrievalError: If an error is encountered while attempting to retrieve the
+                                masque
+        """
+
+    @abstractmethod
+    def set_masque(self, operating_system: OperatingSystem, masque: bytes):
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/repositories/i_masquerade_repository.py:27]:
```suggestion
    def set_masque(self, operating_system: OperatingSystem, masque: Optional[bytes]):
```

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py (hunk 1 above)
Comment (mssalvatore) [reply monkey/agent_plugins/credentials_collectors/ssh/src/plugin.py]: I suppose the type errors will still appear in windows.

We should pull these commits out of this PR and push them to develop.

File: monkey/monkey_island/cc/repositories/i_masquerade_repository.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        """
+        Set the masque for the specified OS.
+
+        :param operating_system: The OS whose masque is to be set
+        :param masque: The value to set the masque to
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/repositories/i_masquerade_repository.py:32]:
```suggestion
        :param masque: The value to set the masque to, or None to unset the masque
```

File: monkey/tests/unit_tests/monkey_island/cc/repositories/test_mongo_masquerade_repository.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+def test_get_masque(masquerade_repository):
+    masque = masquerade_repository.get_masque(OperatingSystem.WINDOWS)
+
+    assert masque == WINDOWS_MASQUE
+
+
+def test_get_masque__empty_masque(masquerade_repository):
+    masque = masquerade_repository.get_masque(OperatingSystem.WINDOWS)
+
+    assert masque == WINDOWS_MASQUE
```
Comment (mssalvatore) [inline monkey/tests/unit_tests/monkey_island/cc/repositories/test_mongo_masquerade_repository.py:96]: This test is identical to the previous one.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**